### PR TITLE
TextEditor: Detect and ask for reload on external file change

### PR DESCRIPTION
**Pull request by LucasChollet**
Edit: I finally added some automatic detection

### CODE REVIEW

**Comment by awesomekling** [comment]
Is this a commonly needed feature? 🤔

**Comment by LucasChollet** [comment]
On its own, not something that I will click on every day, to be honest.
But with automatic detection, it's quite useful when another program modifies the file (ex: git).
I think a lot of editors have this feature.

**Comment by awesomekling** [comment]
Hmm, it seems that this feature is usually called "Revert" in other editors, not "Reload".
I think that might make a better name here too.

**Comment by awesomekling** [comment]
> This branch has conflicts that must be resolved

**Comment by LucasChollet** [comment]
Done!

File: Userland/Libraries/LibGUI/MessageBox.cpp [hunk 1]
```diff
@@ -55,6 +55,25 @@ int MessageBox::ask_about_unsaved_changes(Window* parent_window, StringView path
     return box->exec();
 }
 
+int MessageBox::ask_about_revert_file(Window* parent_window, StringView path)
```
Comment (awesomekling) [inline Userland/Libraries/LibGUI/MessageBox.cpp:58]: This name feels really cryptic. What we're really doing is asking about reloading a file that has changed on disk.
`MessageBox::ask_to_reload_file_that_has_changed_on_disk()` or something like that perhaps?

Comment (LucasChollet) [reply Userland/Libraries/LibGUI/MessageBox.cpp]: Done!

**Pull request closed without merging**